src/main/java/frc/robot/subsystems/HoodSubsystem.java

```diff
@@ -0,0 +1,169 @@
+     * object
+     * is constructed by calling the getPIDController() method on an existing
+     * CANSparkMax object
+     */
+    m_pidController = m_motor.getPIDController();
+
+    // Encoder object created to display position values
+    m_encoder = m_motor.getEncoder();
+
+    // PID coefficients
+    kP = 0.001;
+    kI = 0.00001;
+    kD = 0;
+    kIz = 0;
+    kFF = 0;
+    kMaxOutput = 1;
+    kMinOutput = -1;
+    
+    
+
+    // set PID coefficients
+    m_pidController.setP(kP);
+    m_pidController.setI(kI);
+    m_pidController.setD(kD);
+    m_pidController.setIZone(kIz);
+    m_pidController.setFF(kFF);
+    m_pidController.setOutputRange(kMinOutput, kMaxOutput);
+
+
+    // display PID coefficients on SmartDashboard
+    SmartDashboard.putNumber("P Gain", kP);
+    SmartDashboard.putNumber("I Gain", kI);
+    SmartDashboard.putNumber("D Gain", kD);
+    SmartDashboard.putNumber("I Zone", kIz);
+    SmartDashboard.putNumber("Feed Forward", kFF);
+    SmartDashboard.putNumber("Max Output", kMaxOutput);
+    SmartDashboard.putNumber("Min Output", kMinOutput);
+    SmartDashboard.putNumber("Set Rotations", rotations);
+
+  }
+
+  @Override
+  public void periodic() {
+    // This method will be called once per scheduler run
+    // read PID coefficients from SmartDashboard
+    double p = SmartDashboard.getNumber("P Gain", 0);
+    double i = SmartDashboard.getNumber("I Gain", 0);
+    double d = SmartDashboard.getNumber("D Gain", 0);
+    double iz = SmartDashboard.getNumber("I Zone", 0);
+    double ff = SmartDashboard.getNumber("Feed Forward", 0);
+    double max = SmartDashboard.getNumber("Max Output", 0);
+    double min = SmartDashboard.getNumber("Min Output", 0);
+    //double rotations = SmartDashboard.getNumber("Set Rotations", 0);
+    
+ 
+    // if PID coefficients on SmartDashboard have changed, write new values to
+    // controller
+    if ((p != kP)) {
+      m_pidController.setP(p);
+      kP = p;
+    }
+    if ((i != kI)) {
+      m_pidController.setI(i);
+      kI = i;
+    }
+    if ((d != kD)) {
+      m_pidController.setD(d);
+      kD = d;
+    }
+    if ((iz != kIz)) {
+      m_pidController.setIZone(iz);
+      kIz = iz;
+    }
+    if ((ff != kFF)) {
+      m_pidController.setFF(ff);
+      kFF = ff;
+    }
+    if ((max != kMaxOutput) || (min != kMinOutput)) {
+      m_pidController.setOutputRange(min, max);
+      kMinOutput = min;
+      kMaxOutput = max;
+    }
+      
+    if(Controller.getAButtonPressed()){
+      rotations = 1.2;
+    }
+    if(Controller.getBButtonPressed()){
+      rotations = 1;
+    }
+    if(Controller.getYButtonPressed()){
+      rotations = 0.2;
+    }
+    if(Controller.getXButtonPressed()){
+      rotations = 0;
+    }
+
+      // must be CANsparkMax.controlType not just controlType
+      m_pidController.setReference(rotations, CANSparkMax.ControlType.kPosition);
+      
+      double position = m_encoder.getPosition();
+
+      if(position < 0){
```

Comment:
This is great protection for the min and max values. You will want to put this before the reference point is set in the PID that way the output is filtered before it is put. Additionally you'll want to reference these values from Constants as well.